From 5b7f29a08ada5ac060959e94c503f34ba5650849 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Thu, 6 Jun 2019 15:58:23 +0530
Subject: [PATCH 0200/1921] octeontx2-pf: Compile PF driver only when enabled

Compile PF driver only when enabled in config

Change-Id: I827b5066646dbb7ffa28dc98b5911f9cd526e8d6
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/10587
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/Kconfig  | 1 +
 drivers/net/ethernet/marvell/octeontx2/Makefile | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/Kconfig b/drivers/net/ethernet/marvell/octeontx2/Kconfig
index 31f794965a5d..95b2d2541e95 100644
--- a/drivers/net/ethernet/marvell/octeontx2/Kconfig
+++ b/drivers/net/ethernet/marvell/octeontx2/Kconfig
@@ -28,6 +28,7 @@ config NDC_DIS_DYNAMIC_CACHING
 
 config OCTEONTX2_PF
 	tristate "Marvell OcteonTX2 NIC Physical Function driver"
+	select OCTEONTX2_MBOX
 	depends on PCI && ARM64 && ARM64_LSE_ATOMICS
 	help
 	  This driver supports Marvell's OcteonTX2 Resource Virtualization
diff --git a/drivers/net/ethernet/marvell/octeontx2/Makefile b/drivers/net/ethernet/marvell/octeontx2/Makefile
index 563dc9cd6b36..0064a69e0f72 100644
--- a/drivers/net/ethernet/marvell/octeontx2/Makefile
+++ b/drivers/net/ethernet/marvell/octeontx2/Makefile
@@ -5,4 +5,4 @@
 
 obj-$(CONFIG_OCTEONTX2_MBOX) += af/
 obj-$(CONFIG_OCTEONTX2_AF) += af/
-obj-y			   += nic/
+obj-$(CONFIG_OCTEONTX2_PF) += nic/
-- 
2.31.1

